src/core/connection.c

```diff
@@ -3801,7 +4027,14 @@ QuicConnRecvFrames(
             } else if (Status == QUIC_STATUS_OUT_OF_MEMORY) {
                 return FALSE;
             } else {
-                if (Status != QUIC_STATUS_INVALID_STATE) {
+                if (Status == QUIC_STATUS_VER_NEG_ERROR) {
+                    // TODO: how to keep the datagram around long enough for the queued operation?
```

Comment:
We need to discuss solutions to this, since it's messy to plumb this specific case all the way to `QuicConnRecvDatagrams`
Maybe a flag on a given datagram to indicate it shouldn't be freed, and promising that it'll be returned later?